### DISCUSSION TITLE
Dont charge battery in the morning

### DISCUSSION THREAD

**Discussion by miisch** [mletenay/home-assistant-goodwe-inverter#311] [Q&A] [answered]
Hello, I have a hard time how to prefer not to charge battery in the morning when spot prices are high and start charging later when spot prices are cheap.
I tried eg. to set eco, eco-discharge or eco-charge and max battery in eco to 10% or with discharge limit charging power to 1%, but none of these were useful (didnt work or it was oscillating around minimum battery level .. charging/discharging between 15-17%).
Could that be by Goodwe firmware? **110923** / DSP FW 11.168 / ARM FW 23.241 (GW10K-ET Plus)
![image](https://github.com/user-attachments/assets/074dee87-bd07-436f-bc83-84e479d576df)

**Comment by miisch** [comment]
I may found the possible cause :-/ this integration (for my device) does not do the same as when I set it via SEMS portal.
When I set eco-discharge and 1% in SEMS it works great, exactly as expected - when I check the values in HA from this integration it still shows General mode and different percentage. This integration does not work for me correctly :,( and set who knows what. Do you have any idea why??
My device: Goodwe; GW10K-ET Plus +

**Reply by janchlebek** [reply, marked as answer]
Hi, no idea but I can only say that with my GW10k this works fine (I'm doing exactly that) and I can set it via HA to Eco-discharge with 1% and it behaves as expected. 
Maybe different behavior based on firmware? Which version you have?

**Comment by miisch** [comment]
Wrong behavior was fixed by firmware update, hooray, finally now everything works just fine.
GW10K-ET Plus
Old: 110923 / DSP FW 11.168 / ARM FW 23.241
New:  121228 / DSP FW 12.198 / ARM FW 28.277
